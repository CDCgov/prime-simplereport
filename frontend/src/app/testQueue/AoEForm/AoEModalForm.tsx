import Modal from "react-modal";

import { displayFullName } from "../../utils";
import { globalSymptomDefinitions } from "../../../patientApp/timeOfTest/constants";
import iconClose from "../../../img/close.svg";

import AoEForm from "./AoEForm";

interface AoEModalProps {
  onClose: () => void;
  patient: any;
  loadState?: any;
  saveCallback: (a: any) => Promise<string | void> | void;
}

const AoEModalForm = (props: AoEModalProps) => {
  const { onClose, patient, loadState = {}, saveCallback } = props;

  const symptomsResponse: { [key: string]: boolean } = {};
  globalSymptomDefinitions.forEach(({ value }) => {
    symptomsResponse[value] = false;
  });

  const buttonGroup = (
    <div className="sr-time-of-test-buttons">
      <button
        className="modal__close-button"
        style={{ cursor: "pointer" }}
        onClick={onClose}
      >
        <img className="modal__close-img" src={iconClose} alt="Close" />
      </button>
    </div>
  );

  return (
    <Modal
      isOpen={true}
      style={{
        content: {
          position: "initial",
        },
      }}
      overlayClassName="prime-modal-overlay display-flex flex-align-center flex-justify-center"
      contentLabel="Test questionnaire"
      ariaHideApp={process.env.NODE_ENV !== "test"}
    >
      <div className="display-flex flex-justify">
        <h1 className="font-heading-lg margin-top-05 margin-bottom-0">
          {displayFullName(
            patient.firstName,
            patient.middleName,
            patient.lastName
          )}
        </h1>
        {buttonGroup}
      </div>
      <div className="border-top border-base-lighter margin-x-neg-205 margin-top-1"></div>
      <h2 className="font-heading-lg margin-top-205 margin-bottom-0">
        Test questionnaire
      </h2>
      <AoEForm
        saveButtonText="Continue"
        onClose={onClose}
        patient={patient}
        loadState={loadState}
        saveCallback={saveCallback}
        isModal={true}
        noValidation={true}
      />
    </Modal>
  );
};

export default AoEModalForm;
