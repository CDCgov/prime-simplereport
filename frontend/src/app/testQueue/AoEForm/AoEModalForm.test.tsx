import { render, RenderResult, screen } from "@testing-library/react";
import MockDate from "mockdate";
import userEvent from "@testing-library/user-event";

import AoEModalForm from "./AoEModalForm";


jest.mock("./AoEForm", () => () => <></>);
// jest.mock("react-modal", () => (props: any) => <>{props.children}</>);

describe("AoEModalForm", () => {
  // let component: RenderResult["container"];
  let component: RenderResult;

  // trying to make Modal component render instead of injecting?
  // unnecessary if react-modal is mocked
  // beforeAll(() => {
  //     ReactDOM.createPortal = jest.fn((element, _node) => {
  //         return element;
  //     }) as any;
  // });

  beforeEach(() => {
    MockDate.set("2021-02-06");
    component = render(
      <AoEModalForm
        onClose={jest.fn()}
        patient={{
          internalId: "123",
          gender: "male",
          firstName: "Steve",
          lastName: "Jobs",
        }}
        saveCallback={jest.fn()}
      />
      // ).container;
    );
  });

  describe("on data loaded", () => {
    beforeEach(async () => {
      // load data
      await screen.findByText("Test questionnaire");
    });

    it("renders", async () => {
      expect(component).toMatchSnapshot();
    });
  });
});

describe("experiment", () => {
  it("closes when it should", async () => {
    let mockClose = jest.fn(() => {});
    MockDate.set("2021-02-06");
    render(
      <AoEModalForm
        onClose={mockClose}
        patient={{
          internalId: "123",
          gender: "male",
          firstName: "Steve",
          lastName: "Jobs",
        }}
        saveCallback={jest.fn()}
      />
    );

    await screen.findByText("Test questionnaire");
    // userEvent.click(screen.getByAltText("Close"));
    userEvent.keyboard("{Escape}");
    expect(mockClose).toHaveBeenCalled();
  });
});
