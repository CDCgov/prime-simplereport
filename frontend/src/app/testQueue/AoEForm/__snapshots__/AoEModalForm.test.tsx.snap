// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AoEModalForm on data loaded renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div />
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div
      class="ReactModalPortal"
    >
      <div
        class="ReactModal__Overlay prime-modal-overlay display-flex flex-align-center flex-justify-center"
      >
        <div
          aria-label="Test questionnaire"
          aria-modal="true"
          class="ReactModal__Content"
          role="dialog"
          style="position: initial; top: 40px; left: 40px; right: 40px; bottom: 40px; border: 1px solid #ccc; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 20px;"
          tabindex="-1"
        >
          <main
            aria-labelledby="aoe-form-title"
          >
            <div
              class="display-flex flex-justify"
            >
              <h1
                class="font-heading-lg margin-top-05 margin-bottom-0"
              >
                Jobs, Steve
              </h1>
              <div
                class="sr-time-of-test-buttons"
              >
                <button
                  class="modal__close-button"
                  style="cursor: pointer;"
                >
                  <img
                    alt="Close"
                    class="modal__close-img"
                    src="close.svg"
                  />
                </button>
              </div>
            </div>
            <div
              class="border-top border-base-lighter margin-x-neg-205 margin-top-1"
            />
            <h2
              class="font-heading-lg margin-top-205 margin-bottom-0"
              id="aoe-form-title"
            >
              Test questionnaire
            </h2>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
